apps/emqx_bridge_snowflake/docs/user-guide.md

```diff
@@ -0,0 +1,119 @@
+
+### macOS
+
+- Install unixODBC (e.g. `brew install unixodbc`)
+- [Download and install iODBC](https://github.com/openlink/iODBC/releases/download/v3.52.16/iODBC-SDK-3.52.16-macOS11.dmg)
+- [Download and install the Snowflake ODBC driver](https://sfc-repo.snowflakecomputing.com/odbc/macuniversal/3.3.2/snowflake_odbc_mac_64universal-3.3.2.dmg)
+- Refer to [Installing and configuring the ODBC Driver for macOS](https://docs.snowflake.com/en/developer-guide/odbc/odbc-mac) for more information.
+- Update `~/.odbc.ini` and `/opt/snowflake/snowflakeodbc/lib/universal/simba.snowflake.ini`:
+
+```sh
+chown $(id -u):$(id -g) /opt/snowflake/snowflakeodbc/lib/universal/simba.snowflake.ini
+echo 'ODBCInstLib=libiodbcinst.dylib' >> /opt/snowflake/snowflakeodbc/lib/universal/simba.snowflake.ini
+
+cat < EOF > ~/.odbc.ini
+[ODBC]
+Trace=no
+TraceFile=
+
+[ODBC Drivers]
+Snowflake = Installed
+
+[ODBC Data Sources]
+snowflake = Snowflake
+
+[Snowflake]
+Driver = /opt/snowflake/snowflakeodbc/lib/universal/libSnowflake.dylib
+EOF
+```
+
+## Create user account and database
+
+After completing the steps below, the following credentials shall be used in connector in action:
+
+| Field                  | Value                                            |
+|------------------------|--------------------------------------------------|
+| Data Source Name(DSN)  | `snowflake`                                      |
+| Username               | `snowpipeuser`                                   |
+| Password               | `Snowpipeuser99`                                 |
+| Database Name          | `testdatabase`                                   |
+| Schema                 | `public`                                         |
+| Stage                  | `teststage0`                                     |
+| Pipe                   | `testpipe0`                                      |
+| Pipe User              | `snowpipeuser`                                   |
+| Private Key            | `file://<path to snowflake_rsa_key.private.pem>` |
+
+### Generate RSA key pair
+
+```sh
+openssl genrsa 2048 | openssl pkcs8 -topk8 -inform PEM -out snowflake_rsa_key.private.pem -nocrypt
+openssl rsa -in snowflake_rsa_key.private.pem -pubout -out snowflake_rsa_key.public.pem
+```
+
+### Snowflake SQL Worksheet (+ Create --> SQL Worksheet)
+    
+```sql
+USE ROLE accountadmin;
+
+CREATE DATABASE IF NOT EXISTS testdatabase;
+
+CREATE OR REPLACE TABLE testdatabase.public.test0 (
+    clientid STRING,
+    topic STRING,
+    payload STRING,
+    publish_received_at TIMESTAMP_LTZ
+);
+
+CREATE STAGE IF NOT EXISTS testdatabase.public.teststage0
+FILE_FORMAT = (TYPE = CSV PARSE_HEADER = TRUE)
+COPY_OPTIONS = (ON_ERROR = CONTINUE PURGE = TRUE);
+
+CREATE PIPE IF NOT EXISTS testdatabase.public.testpipe0 AS
+COPY INTO testdatabase.public.test0
+FROM @testdatabase.public.teststage0
+MATCH_BY_COLUMN_NAME = CASE_INSENSITIVE;
+
+CREATE OR REPLACE WAREHOUSE testwarehouse;
```

Comment:
This is used only by the test user to select stuff; when using the snowflake worksheets, I think they create a default `computewh` one